public/app/features/explore/state/utils.ts

```diff
@@ -63,7 +63,17 @@ export async function loadAndInitDatasource(
   orgId: number,
   datasourceName?: string
 ): Promise<{ history: HistoryItem[]; instance: DataSourceApi }> {
-  const instance = await getDatasourceSrv().get(datasourceName);
+  let instance;
+  try {
+    instance = await getDatasourceSrv().get(datasourceName);
+  } catch (error) {
+    // Falling back to the default data source in case the provided data source was not found.
+    // It may happen if last used data source or the data source provided in the URL has been
+    // removed or it is not provisioned anymore.
+    console.error(error);
+    instance = await getDatasourceSrv().get();
+    console.info('Default datasource loaded.');
```

Comment:
do we really need to print to console?

Reply:
Nope and hmmm... actually in this case might be even misleading as both lines may not appear next to each other. I'll remove `console.info`. I think it may be worth keeping `console.error` as there is no UI warning displayed in this case.

Reply:
but what's the point of showing an error in the console in the first place?

if this was an actual runtime error we probably should have surfaced it in the UI, but we actually expect the datasource to not be there, and we just fallback to the default one in case the one you last visited is not present anymore.

Also,

Reply:
True that! I'll ditch both.